### ISSUE TITLE
Connections: createSynapse() has non-optimal / unneeded if test

### ISSUE THREAD

**Issue by cogmission** [numenta/nupic#3285]
A small optimization for clarity's sake and optimal performance:

The [if clause here](https://github.com/numenta/nupic/blob/master/src/nupic/research/connections.py#L423) should be removed and the appending of the synapse should be added above in the ```else``` clause.

There is no reason to perform a second check - the Synapse should be added to the presynapticCell's list of synapses as it is in the above ```if``` part of the if/else clause above.

**Comment by rhyolight** [comment]
@numenta/nupic-reviewers Can someone confirm David's findings before I approve this for work?

**Comment by mrcslws** [comment]
At that point in code we haven't instantiated the `Synapse` yet.

We could create a `Synapse(-1, segment, None)` and then populate its `synapse.idx` and `synapse.data` in the if/else. Then we could make your suggested change.

I agree that it'd be more readable if we change `self._synapsesForSegment` in the `else`, so that it's consistent with where we change it in the `if`.

**Comment by cogmission** [comment]
@mrcslws I agree; and my main focus was on reducing the number of ```if/then``` branches - one less is always better if the code can be structured to limit those...
